SearchActivity.py

```diff
@@ -187,142 +180,64 @@ def _write_scores_to_clipboard(self, button=None):
 
     # Collaboration-related methods
 
-    def _setup_presence_service(self):
-        """ Setup the Presence Service. """
-        self.pservice = presenceservice.get_instance()
-        self.initiating = None  # sharing (True) or joining (False)
+    def set_data(self, data):
+        pass
 
-        owner = self.pservice.get_owner()
-        self.owner = owner
-        self._share = ""
-        self.connect('shared', self._shared_cb)
-        self.connect('joined', self._joined_cb)
+    def get_data(self):
+        return None
 
     def _shared_cb(self, activity):
         """ Either set up initial share..."""
-        self._new_tube_common(True)
+        self.after_share_join(True)
 
     def _joined_cb(self, activity):
         """ ...or join an exisiting share. """
         self._new_tube_common(False)
 
     def _new_tube_common(self, sharer):
```

Comment:
You can delete this method since you're no longer using it and it'll be better to handle anything that you need to handle when the activity is joined in the `_joined_cb` rather than having a method that calls another method that calls another method to perform an action when that can be handled in the first method.

Reply:
Same thing also happens in `game.py`, you can eliminate the need for those methods by just having `collab.post` calls, one example is [this](https://github.com/sugarlabs/cookie-search-activity/blob/master/game.py#L158) where `send_new_game` is called and that in turn calls `send_event`.